gallery/gallery/lib/demos/material/bottom_app_bar_demo.dart

```diff
@@ -0,0 +1,170 @@
+      ),
+      body: ListView(
+        padding: const EdgeInsets.only(bottom: 88),
+        children: [
+          SwitchListTile(
+            title: Text(
+              GalleryLocalizations.of(context).demoFloatingButtonTitle,
+            ),
+            value: _showFab,
+            onChanged: _onShowFabChanged,
+          ),
+          SwitchListTile(
+            title: Text(GalleryLocalizations.of(context).bottomAppBarNotch),
+            value: _showNotch,
+            onChanged: _onShowNotchChanged,
+          ),
+          Padding(
+            padding: const EdgeInsets.all(16),
+            child: Text(GalleryLocalizations.of(context).bottomAppBarPosition),
+          ),
+          RadioListTile<FloatingActionButtonLocation>(
+            title: Text(
+              GalleryLocalizations.of(context).bottomAppBarPositionDockedEnd,
+            ),
+            value: FloatingActionButtonLocation.endDocked,
+            groupValue: _fabLocation,
+            onChanged: _onFabLocationChanged,
+          ),
+          RadioListTile<FloatingActionButtonLocation>(
+            title: Text(
+              GalleryLocalizations.of(context).bottomAppBarPositionDockedCenter,
+            ),
+            value: FloatingActionButtonLocation.centerDocked,
+            groupValue: _fabLocation,
+            onChanged: _onFabLocationChanged,
+          ),
+          RadioListTile<FloatingActionButtonLocation>(
+            title: Text(
+              GalleryLocalizations.of(context).bottomAppBarPositionFloatingEnd,
+            ),
+            value: FloatingActionButtonLocation.endFloat,
+            groupValue: _fabLocation,
+            onChanged: _onFabLocationChanged,
+          ),
+          RadioListTile<FloatingActionButtonLocation>(
+            title: Text(
+              GalleryLocalizations.of(context)
+                  .bottomAppBarPositionFloatingCenter,
+            ),
+            value: FloatingActionButtonLocation.centerFloat,
+            groupValue: _fabLocation,
+            onChanged: _onFabLocationChanged,
+          ),
+        ],
+      ),
+      floatingActionButton: _showFab
+          ? FloatingActionButton(
+              onPressed: () {
+                print('Floating action button pressed');
+              },
+              child: Icon(Icons.add),
+              tooltip: GalleryLocalizations.of(context).buttonTextCreate,
+            )
+          : null,
+      floatingActionButtonLocation: _fabLocation,
+      bottomNavigationBar: _DemoBottomAppBar(
+        fabLocation: _fabLocation,
+        shape: _showNotch ? CircularNotchedRectangle() : null,
+      ),
+    );
+  }
+}
+
+class _DemoBottomAppBar extends StatelessWidget {
+  const _DemoBottomAppBar({
+    this.fabLocation,
+    this.shape,
+  });
+
+  final FloatingActionButtonLocation fabLocation;
+  final NotchedShape shape;
+
+  static final List<FloatingActionButtonLocation> kCenterLocations =
```

Comment:
nit: don't need this type, also, don't use `k` as a prefix.

https://dart.dev/guides/language/effective-dart/style#dont-use-prefix-letters

```suggestion
  static final centerLocations =
```

Reply:
done